2_base_and_sensor.ipynb

```diff
@@ -436,135 +812,259 @@
     "\n",
     "センサ値の単位はメートルです。"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Since each observation point on the scan range (-120 degrees to 120 degrees) is digitized from left to right, the element in the middle of the array (= 361th) represents the \"distance from the front of the robot to the wall\".\n",
+    "\n",
+    "The unit of sensor value is meters."
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "outputs": [],
    "source": [
-    "# 361番目のデータにアクセス\n",
+    "# 361番目のデータにアクセス Access the 361th data\n",
     "scan_data.ranges[360]"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "# センサ値を使ってロボットを制御する"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "# Control the robot using sensor values"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "センサ値を使ってロボットを制御してみましょう。"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Let's control the robot using the sensor value."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "壁の1メートル手前で止まる（壁の1メートル手前まで前進する）プログラムを書いてみましょう。\n",
     "\n",
     "while文を使うと以下のように書けると思います。"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Let's write a program that stops 1 meter before the wall (go straight until the robot reaches 1meter before the wall).\n",
+    "\n",
+    "You can use `while` statement to write your code:"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "outputs": [],
    "source": [
     "while True:\n",
     "    scan_data = laser.get_data()\n",
-    "    if scan_data.ranges[360] < 1.0:  # 1メートル以内になったらwhileループを抜ける\n",
+    "    if scan_data.ranges[360] < 1.0:  # 1メートル以内になったらwhileループを抜ける Exit the while loop if within 1 meter\n",
     "        break\n",
     "        \n",
     "    move_base(0.5, 0, 0)"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "<div class=\"alert alert-block alert-info\">\n",
     "    <b>課題</b>\n",
     "    <p>初期位置から人の前まで移動するプログラムを書いてみましょう。今回はレーザの値を用いましょう。</p>\n",
     "</div>"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "<div class=\"alert alert-block alert-info\">\n",
+    "    <b>Task</b>\n",
+    "    <p>Let's write a code to move the robot from the initial position to the front of the human.</p>\n",
```

Comment:
In both cases below, I replaced "a code" with "some code".
When referring to code (as in programming code or source code), we can never say "a code".
"A code" refers to something different. :)

The following sentence uses "the human".  This means that there is only one human and that we know about this human from a prior context or that there is more than one human and from prior context we can be sure of which human we are referring to.

    "    <p>Let's write some code to move the robot from the initial position to the front of the human.</p>\n",

The following sentence uses "a human".   This suggests that there is more than one human.

    "    <p>Let's write some code to move the robot from the initial position to the front of a human.</p>\n",